doc/newfeat.texi

```diff
@@ -28,11 +22,35 @@ from capturing the cursor.
 @item @ESS{[R]}: New command @code{ess-r-package-use-dir}. It sets the
 working directory of the current process to the current package directory.
 
-@item @ESS{[R]}: Launching or reloading R from a package file now sets
-the working directory to the package path. You can now launch R from
-files within @code{R/} or @code{src/} without worrying about the wd.
+@item @ESS{[R]}: Launching R from a package file now recognises the package
+root as the likely startup directory for R. If
+@code{ess-ask-for-ess-directory} is non-nil, ESS prompts with the
+package root as default selection. Otherwise the working directory is
+automatically set to the package root.
+
+The mechanism is a bit more general if you are using Emacs 25.1 or
+older. On those recent Emacsen R packages are now recognised as Emacs
+projects and ESS queries Emacs projects to get the default startup
+directory. This means that any git or subversion directory will be
+recognised as project root even if it is not an R package.
+
+In light of these developments, @code{ess-r-package-get-info} is
```

Comment:
Need not be in the newfeat, I think. `ess-r-package-get-info` is kinda an internal function, so I think we can just silently deprecate it. Discussion of this feature got a bit too technical.